pkgs/applications/editors/lite-xl/default.nix

```diff
@@ -0,0 +1,129 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, SDL2
+, lua52Packages
+, pkg-config
+, makeWrapper
+, openlibm
+, meson
+, pcre2
+, freetype
+, git
+, reproc
+, agg
+, ninja
+, curl
+, unzip
+, writeScriptBin
+}:
+
+stdenv.mkDerivation rec {
+  pname = "lite-xl";
+  version = "2.0.1";
+
+  src = fetchFromGitHub {
+    owner = "lite-xl";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-+RbmT6H/5Ldhv3qOClxMjCSGMudbkGtkjo2SpGqExao=";
+  };
+
+  nativeBuildInputs = [ makeWrapper pkg-config meson ninja curl unzip ];
+
+  buildInputs = [ SDL2 lua52Packages.lua openlibm pcre2 freetype git agg reproc ];
+
+  patches = [
+    # this patch makes meson fallback to local dependency instead of fetching it.
+    ./meson-fallback.patch
+    # this patch replaces unpack() with table.unpack(). this should be removed when the next release is proposed (2.0.2). Check PR #435.
+    ./table-unpack.patch
+  ];
+
+  # Nix automatically tries to run meson
+  dontUseMesonConfigure = true;
+
+  buildPhase = ''
+    get_os_kind() {
+      if [[ "$OSTYPE" == "darwin"* ]]; then
+        echo "macos"
+      elif [[ "$OSTYPE" == "linux"* || "$OSTYPE" == "freebsd"* ]]; then
+        echo "linux"
+      else
+        echo "UNSUPPORTED-OS"
+      fi
+    }
+
+    main() {
+      local os=$(get_os_kind)
+      if [[ $os == "UNSUPPORTED-OS" ]]; then
+        echo "ERR: unknown OS type: \"$OSTYPE\""
+        exit 1
+      fi
+
+      local arch=$(uname -m)
+      local build_dir=build
+      local dest_dir=lite-xl
+      local prefix=/
+      local debug
+      local force_fallback
+
+      [[ -d "$build_dir" ]] && rm -rf "$build_dir"
+
+      meson setup \
+        --prefix "$prefix" \
+        $force_fallback \
+        "$build_dir"
+
+      meson compile -C "$build_dir"
+
+      local data_dir="$(pwd)/$dest_dir/$prefix/share/lite-xl"
+      local executable="$(pwd)/$dest_dir/$prefix/bin/lite-xl"
+      local stripcmd="strip"
+
+      # Clean up previous package and directory
+      [[ -d "$dest_dir" ]] && rm -rf "$dest_dir"
+      [[ -f "$dest_dir.tar.gz" ]] && rm "$dest_dir".tar.gz
+      [[ -f "$dest_dir.zip" ]] && rm "$dest_dir".zip
+      mkdir -p "$data_dir"
+
+      # Copy third party modules
+      cp ${builtins.fetchurl {
```

Comment:
We probably don't want this to put directly in buildPhase. Check how other packages are doing this (`hddtemp` for example).

Reply:
yeah, just checked ofborg's error. Is there any way that I can locally evaluate default.nix in restricted mode?